libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -199,25 +181,26 @@ proc getGossipPeers(g: GossipSub): Table[string, ControlMessage] {.gcsafe.} =
           result[id].ihave.add(ihave)
 
 proc heartbeat(g: GossipSub) {.async.} =
-  try:
-    await g.heartbeatLock.acquire()
-    trace "running heartbeat"
+  while true:
+    try:
+      await g.heartbeatLock.acquire()
+      trace "running heartbeat"
 
-    await sleepAsync(GossipSubHeartbeatInitialDelay)
+      for t in g.mesh.keys:
+        await g.rebalanceMesh(t)
 
-    for t in g.mesh.keys:
-      await g.rebalanceMesh(t)
+      await g.dropFanoutPeers()
+      let peers = g.getGossipPeers()
+      for peer in peers.keys:
+        await g.peers[peer].send(@[RPCMsg(control: some(peers[peer]))])
```

Comment:
shouldn't there be a try/catch around here? a single peer failing should not cause the rest of the peers to not receive the message

Reply:
actually, an `allFutures`/`allFinished` might be the right thing here?

Reply:
but, yeah shouldn't stop broadcasting.

Reply:
`allFinished` and warn if error I guess is the best here